site/docs/v1/tech/apis/message-templates.adoc

```diff
@@ -0,0 +1,216 @@
+--
+
+==== Request Parameters
+
+[.api]
+[field]#messageTemplateId# [type]#[UUID]# [optional]#Optional#::
+The Id of the Message Template to retrieve.
+
+=== Response
+
+The response for this API contains either a single Message Template or all of the Message Templates. When you call this API with an Id the response will contain just that Message Template. When you call this API without an Id the response will contain all of the Message Templates. Both response types are defined below along with an example JSON response.
+
+:no_errors:
+:never_search_error:
+include::docs/v1/tech/apis/_standard-get-response-codes.adoc[]
+:no_errors!:
+:never_search_error!:
+
+include::docs/v1/tech/apis/_message-template-response-body.adoc[]
+
+include::docs/v1/tech/apis/_message-templates-response-body.adoc[]
+
+
+== Update a Message Template
+
+This API is used to update a Message Template. You must specify the Id of the Message Template to update on the URI.
+
+=== Request
+
+[.api-authentication]
+link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:lock[type=fas]] Update a Message Template by Id
+[.endpoint]
+.URI
+--
+[method]#PUT# [uri]#/api/message/template/``\{messageTemplateId\}``#
+--
+
+:api_endpoint: pass:normal[/api/message/template/``\{messageTemplateId\}``]
+include::docs/v1/tech/apis/_patch-support.adoc[]
+
+==== Request Parameters
+
+[.api]
+[field]#messageTemplateId# [type]#[UUID]# [required]#Required#::
+The Id of the Message Template to update.
+
+include::docs/v1/tech/apis/_message-template-request-body.adoc[]
+
+=== Response
+
+The response for this API contains the new information for the Message Template that was updated.
+
+:never_search_error:
+include::docs/v1/tech/apis/_standard-put-response-codes.adoc[]
+:never_search_error!:
+
+include::docs/v1/tech/apis/_message-template-response-body.adoc[]
+
+
+== Delete a Message Template
+
+This API is used to delete a Message Template. You must specify the Id of the Message Template on the URI.
+
+=== Request
+
+[.api-authentication]
+link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:lock[type=fas]] Delete a Message Template By Id
+[.endpoint]
+.URI
+--
+[method]#DELETE# [uri]#/api/message/template/``\{messageTemplateId\}``#
+--
+
+==== Request Parameters
+
+[.api]
+[field]#messageTemplateId# [type]#[UUID]# [required]#Required#::
+The Id of the Message Template to delete.
+
+=== Response
+
+This API does not return a JSON response body.
+
+:never_search_error:
+include::docs/v1/tech/apis/_standard-delete-response-codes.adoc[]
+:never_search_error!:
+
+
+== Preview a Message Template
+
+This API is used to preview a Message Template. You pass all of the information for the Message Template in the request and a rendered version of the Message is sent back to you in the response. The Message Template in the request does not need to be completely filled out. You can send in a partial Message Template and the response will contain only what you provided.
```

Comment:
Do you think it would be useful to note that we mock the value for ${code} and ${user} when previewing?

Code is set to `123456` as shown in the response.json.  The user object that is mocked is fairly large and can be found `src/main/java/io/fusionauth/app/action/api/message/template/PreviewAction.java:71-104`.